File: Android/doric/src/main/java/com/github/penfeizhou/doric/shader/StackNode.java
package com.github.penfeizhou.doric.shader;

import android.widget.FrameLayout;

import com.github.penfeizhou.doric.DoricContext;
import com.github.penfeizhou.doric.extension.bridge.DoricPlugin;
import com.github.pengfeizhou.jscore.JSObject;
import com.github.pengfeizhou.jscore.JSValue;

/**
 * @Description: com.github.penfeizhou.doric.widget
 * @Author: pengfei.zhou
 * @CreateDate: 2019-07-20
 */
@DoricPlugin(name = "Stack")
public class StackNode extends GroupNode<FrameLayout> {
    public StackNode(DoricContext doricContext) {
        super(doricContext);
    }

    @Override
    public FrameLayout build(JSObject jsObject) {
        return new FrameLayout(getContext());
    }

    @Override
    protected void blend(FrameLayout view, String name, JSValue prop) {
        switch (name) {
            case "gravity":
                view.setForegroundGravity(prop.asNumber().toInt());
                break;
            default:
                super.blend(view, name, prop);
        }
    }
}

File: doric-flutter/doric/lib/doric/shader/stack_node.dart
import 'package:flutter/cupertino.dart';
import 'package:doric/doric/doric_context.dart';
import 'package:doric/doric/layout/doric_node_data.dart';
import 'package:doric/doric/layout/doric_widget.dart';
import 'package:doric/doric/shader/group_node.dart';

class StackNode<T extends Widget> extends GroupNode<Widget> {
  StackNode(DoricContext context) : super(context);

  @override
  Widget build(Map props) {
    return DoricStack(data:DoricNodeData(props),children: this.getChildWidgets(),);
  }
}

